src/backend/sgx/ioctls.rs

```diff
@@ -152,6 +156,45 @@ impl RestrictPermissions {
     pub fn count(&self) -> u64 {
         self.count
     }
+
+    pub fn execute(&self, mutex_fd: &Mutex<fs::File>) -> io::Result<()> {
+        let mut count: u64 = 0;
+        while self.length > count {
+            let mut parameters = Self {
+                offset: self.offset + count,
+                length: self.length - count,
+                permissions: self.permissions,
+                result: 0,
+                count: 0,
+            };
+
+            let fd_locked = mutex_fd.lock().unwrap();
+            let mut fd_cloned = fd_locked.try_clone().unwrap();
+            match ENCLAVE_RESTRICT_PERMISSIONS.ioctl(&mut fd_cloned, &mut parameters) {
+                Ok(_) => {}
+                // EINTR
+                Err(e) if matches!(e.raw_os_error(), Some(4)) => {
```

Comment:
Should this magic number be a constant from somewhere?

Reply:
right.. I could get them from `libc`

Reply:
replaced